pkg/docker/config/config.go

```diff
@@ -3,58 +3,94 @@ package config
 import (
 	"encoding/base64"
 	"encoding/json"
+	"fmt"
 	"strings"
-
-	"github.com/pkg/errors"
 )
 
 // The following types are copied from the Kubernetes codebase,
 // since it is not placed in any of the officially supported client
 // libraries.
-// See https://github.com/kubernetes/kubernetes/blob/v1.31.1/pkg/credentialprovider/config.go
+// See https://github.com/kubernetes/kubernetes/blob/v1.33.1/pkg/credentialprovider/config.go
 
 // DockerConfigJSON represents ~/.docker/config.json file info
 // see https://github.com/docker/docker/pull/12009.
 type DockerConfigJSON struct {
-	Auths DockerConfig `json:"auths"`
+	Auths       DockerConfig      `json:"auths"`
+	HTTPHeaders map[string]string `json:"HttpHeaders,omitempty"`
```

Comment:
nit: I think we can change the "copied" from `// The following types are copied from the Kubernetes codebase,` to "adapted" and we can remove the `HTTPHeaders` field, as we don't use it